protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -493,7 +493,9 @@ func renderMessageAsDefinition(msg *descriptor.Message, reg *descriptor.Registry
 		}
 
 		if fieldSchema.Required != nil {
+			schema.Required = searchRequired(schema.Required, fieldSchema.Required)
 			schema.Required = append(schema.Required, fieldSchema.Required...)
+			fieldSchema.Required = nil
```

Comment:
Could you add a comment to explain what's happening here? I don't understand why this search is required.